File: rb/features.rb
# ImagePlaceholderGenerator SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ImagePlaceholderGeneratorFeatures
  def self.make_feature(name)
    case name
    when "base"
      ImagePlaceholderGeneratorBaseFeature.new
    when "ratelimit"
      ImagePlaceholderGeneratorRatelimitFeature.new
    when "retry"
      ImagePlaceholderGeneratorRetryFeature.new
    when "test"
      ImagePlaceholderGeneratorTestFeature.new
    when "timeout"
      ImagePlaceholderGeneratorTimeoutFeature.new
    else
      ImagePlaceholderGeneratorBaseFeature.new
    end
  end
end

File: py/features.py
# ImagePlaceholderGenerator SDK feature factory

from feature.base_feature import ImagePlaceholderGeneratorBaseFeature
from feature.test_feature import ImagePlaceholderGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ImagePlaceholderGeneratorBaseFeature(),
        "test": lambda: ImagePlaceholderGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
